### PR TITLE
DAOS-11468 gurt, vos, pool: fixed several coverity issues

### PR DESCRIPTION
**Pull request by wiliamhuang**
128141, 128142, 128144, 128872, 128873, 128874

Required-githooks: true

Signed-off-by: Lei Huang <lei.huang@intel.com>

### CODE REVIEW

**Comment by wiliamhuang** [comment]
All CI tests passed.

File: src/gurt/dlog.c [hunk 1]
```diff
@@ -936,6 +936,11 @@ d_log_open(char *tag, int maxfac_hint, int default_mask, int stderr_mask,
 			fprintf(stderr, "d_log_write(): failed to recreate log file %s: %s\n",
 				mst.log_file, strerror(errno));
 			close(mst.log_fd);
+			clog_unlock();
+			if (buffer)
+				free(buffer);
+			if (newtag)
+				free(newtag);
```
Comment (daltonbohning) [inline src/gurt/dlog.c:943]: Could we just `goto error` here?

File: src/vos/vos_pool_scrub.c [hunk 2]
```diff
@@ -379,8 +379,10 @@ sc_verify_recx(struct scrub_ctx *ctx, d_iov_t *data)
 		bool		 match;
 		daos_size_t	 rec_in_chunk;
 
-		if (sc_cont_is_stopping(ctx))
+		if (sc_cont_is_stopping(ctx))	{
+			D_FREE(csum_buf);
 			return 0;
+		}
```
Comment (daltonbohning) [inline src/vos/vos_pool_scrub.c:385]: Could we `goto done`?

**Review by daltonbohning — APPROVED** [review]
LGTM, but suggested consistency

File: src/vos/vos_pool_scrub.c (hunk 2 above)
Comment (wiliamhuang) [reply src/vos/vos_pool_scrub.c:385]: Good suggestion. I will update it. Thank you!

Comment (wiliamhuang) [reply src/vos/vos_pool_scrub.c]: Fixed. Thank you!

**Review by daltonbohning — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Review by wangshilong — APPROVED** [review]

**Comment by wiliamhuang** [comment]
@jolivier23 Can you land this PR? Thank you!

**Pull request merged**